tracer/src/Datadog.Trace/Agent/Transports/ApiWebRequest.cs

```diff
@@ -46,5 +48,71 @@ public async Task<IApiResponse> PostAsync(ArraySegment<byte> bytes, string conte
                 return new ApiWebResponse((HttpWebResponse)exception.Response);
             }
         }
+
+        public async Task<IApiResponse> PostAsync(params MultipartFormItem[] items)
+        {
+            if (items is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(nameof(items));
+            }
+
+            Log.Debug<int>("Sending multipart form request with {Count} items.", items.Length);
+
+            var boundary = "---------------------------" + DateTime.Now.Ticks.ToString("x");
+
+            _request.Method = "POST";
+            _request.ContentType = "multipart/form-data; boundary=" + boundary;
+
+            using (var requestStream = await _request.GetRequestStreamAsync().ConfigureAwait(false))
+            {
+                // Write form request using the boundary
+
+                var boundaryBytes = Encoding.ASCII.GetBytes("\r\n--" + boundary + "\r\n");
```

Comment:
Is the "initial" `\r\n` here correct? 🤔 I assume the request body already has the two `\r\n` necessary to separate the headers from the body, so this would add a third `\r\n` for the first item?

Reply:
The initial one is not necessary for the first item, but starting index 1, we need to write those `\r\n` to separete the boundary from the previous form item body. I left it like this because in my tests this additional `\r\n` didn't break anything and just created one allocation for the whole boundary separator line.

Reply:
>  I left it like this because in my tests this additional \r\n didn't break anything 

😬 That's what I was afraid of... I just don't trust that we've trusted _everything_ to not break. Having a custom Http client scares me enough, _knowingly_ not following the spec, more-so 😅